torchtext/prototype/generate.py

```diff
@@ -0,0 +1,110 @@
+from typing import Optional
+
+import torch
+import torch.nn.functional as F
+from torch import nn
+
+
+class GenerationUtil:
+    """Wrapper to provide generation utils for encoder/decoder models and decoder models.
+
+    Example:
+    >>> model = T5_BASE_GENERATION.get_model()
+    >>> generative_model = GenerationUtil(model=model, is_encoder_decoder=True)
+    >>> generative_model.generate(input_ids, num_beams=1, max_len=100)
+    """
+    def __init__(self, model: nn.Module, is_encoder_decoder: bool = True) -> None:
+        self.model = model
+        self.is_encoder_decoder = is_encoder_decoder
+
+    def _prepare_decoder_ids_for_generation(
+        self, batch_size: int, device: Optional[torch.device] = None, **model_kwargs
+    ):
+        if model_kwargs is not None and "decoder_input_ids" in model_kwargs:
+            return model_kwargs.pop("decoder_input_ids")
```

Comment:
why pass around model_kwargs dict instead of just having an optional decoder_input_ids param?